### PR TITLE
doc: add Python keyring to Troubleshooting documentation

### PR DESCRIPTION
**Pull request by tobitege**
During build process some Python dependencies could get stuck in "pending" mode if `PYTHON_KEYRING_BACKEND` is not set in env (Poetry had a couple of issues related to this on their site, e.g. https://github.com/python-poetry/poetry/issues/5250).

Probably fixes: https://github.com/OpenDevin/OpenDevin/issues/919, https://github.com/OpenDevin/OpenDevin/issues/2285

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,11 @@
 SHELL=/bin/bash
 # Makefile for OpenDevin project
 
+# Check and set PYTHON_KEYRING_BACKEND if not already set
+ifndef PYTHON_KEYRING_BACKEND
+export PYTHON_KEYRING_BACKEND=keyring.backends.null.Keyring
+endif
```
Comment (li-boxuan) [inline Makefile:7]: If poetry doesn't want to include this, it seems we shouldn't do either. IMO a better place for this is the troubleshooting guide: https://opendevin.github.io/OpenDevin/modules/usage/troubleshooting

Comment (tobitege) [reply Makefile:7]: hmm... not sure that Poetry itself *should* include it anywhere, they just have to support it (for credentials).
Did you find a post of theirs about stating that, though?

However, I'll revert that and add it to the troubleshooter. 😀

Comment (li-boxuan) [reply Makefile:7]: They probably fixed it (after tons of complaints, thumbs up for workaround, and thumbs down for maintainers XD): https://github.com/python-poetry/poetry/issues/1917#issuecomment-1745782011

but someone still reported the same issue: https://github.com/python-poetry/poetry/issues/1917#issuecomment-1793322478

Comment (li-boxuan) [reply Makefile:7]: So it's either they don't want to fix, or they don't have bandwidth to fix. I am generally opposed to applying hacks on our side if upstream doesn't fix it. Troubleshooting guide is the optimal place IMO, but others might have different opinions.

Comment (tobitege) [reply Makefile:7]: Well, they fixed to recognize 2 different settings of the keyring. Not sure what "fix" you expect?
Not sure they should just use some default value themselves.
Anyhow, already updating the documentation and hopefully then less users will run into this. ;)

Comment (tobitege) [reply Makefile]: Ready for review

File: docs/modules/usage/troubleshooting/troubleshooting.md [hunk 2]
```diff
@@ -62,17 +63,19 @@ especially Windows, this seems to fail.
 
 ### Workarounds
 
-- Restart your computer (sometimes works?)
-- Be sure to have the latest versions of WSL and Docker
-- Try [this reinstallation guide](https://github.com/OpenDevin/OpenDevin/issues/1156#issuecomment-2064549427)
-- Set `-e SANDBOX_TYPE=exec` to switch to the ExecBox docker container
+* Restart your computer (sometimes it does works)
```
Comment (li-boxuan) [inline docs/modules/usage/troubleshooting/troubleshooting.md:66]:
```suggestion
* Restart your computer (sometimes it does work)
```

File: docs/modules/usage/troubleshooting/troubleshooting.md [hunk 3]
```diff
@@ -83,18 +86,20 @@ AttributeError: 'NoneType' object has no attribute 'request'
...
-This usually happens with local LLM setups, when OpenDevin can't connect to the LLM server.
+This usually happens with *local* LLM setups, when OpenDevin can't connect to the LLM server.
 See our guide for [local LLMs](llms/localLLMs) for more information.
 
 ### Workarounds
 
-- Check your `LLM_BASE_URL`
-- Check that ollama is running OK
-- Make sure you're using `--add-host host.docker.internal:host-gateway` when running in docker
+* Check your `LLM_BASE_URL` in your config.toml (if it exists) or command line call
```
Comment (li-boxuan) [inline docs/modules/usage/troubleshooting/troubleshooting.md:94]: not accurate; it's `base_url` under an LLM config group in config.toml

**Review by li-boxuan — APPROVED** [review]
thanks, just some minor issues and I think we are good to go!

File: docs/modules/usage/troubleshooting/troubleshooting.md (hunk 3 above)
Comment (tobitege) [reply docs/modules/usage/troubleshooting/troubleshooting.md:94]: Oh, thanks for that, fixed here.

But wanted to ask for your feedback about another entry I'd like to add, but I'm not quite yet sure if it is actually 2 separate entries that touch sessions (local storage, jwt secret) and how to word it best:

## Sessions are not restored

### Symptoms

OpenDevin usually asks whether to resume or start a new session when opening the UI.
But clicking "Resume" still starts a fresh new chat.

### Details

With a standard installation as of today session data is stored in memory.
Currently, if OpenDevin's service is restarted, previous sessions become
invalid (a new secret is generated) and thus not recoverable.

### Workarounds

* Change configuration to make sessions persistent by editing the `config.toml`
file (in OpenDevin's root folder) by specifying a `file_store` and an
absolute `file_store_path`:

```toml
file_store="local"
file_store_path="/absolute/path/to/opendevin/cache/directory"
```

**Comment by tobitege** [comment]
Added a short hint to Makefile for setup config to mention absolute path (solves #2289 https://github.com/OpenDevin/OpenDevin/pull/2289)

**Comment by tobitege** [comment]
Added a short hint to Makefile for setup config to mention absolute path (solves #2246)

**Pull request merged**